network/address_test.go

```diff
@@ -458,9 +478,19 @@ var selectInternalMachineTests = []selectTest{{
 		network.NewScopedAddress("fe80::1", network.ScopeLinkLocal),
 		network.NewScopedAddress("127.0.0.2", network.ScopeMachineLocal),
 	},
+	4,
+}, {
+	"first machine local non-IPv4 address is selected even with public/cloud hostnames",
```

Comment:
There has to be no IPv4 addresses or they will be selected in preference - hence this is a test for "non IPv4 address" selection.